src/utils/fetch.ts

```diff
@@ -10,9 +10,13 @@ export const getFetch = (config: PineconeConfiguration) => {
     // scope, use that. This should prevent confusing failures in
     // nextjs projects where @vercel/fetch is mandated and
     // other implementations are stubbed out.
+    console.log(
+      'Failed to find any user-provided fetch implementation. Using global fetch implementation.'
+    );
     return global.fetch;
   } else {
-    // Use ponyfill as last resort
-    return crossFetch;
+    throw new PineconeConfigurationError(
+      'No global or user-provided fetch implementations found. Please supply a fetch implementation.'
+    );
```

Comment:
I think throwing in this specific instance is fine. I'm trying to think of any unintended consequences from removing the generic fallback.

Are there possible instances where people were relying on `crossFetch` previously and now we're pulling it out? I don't think this would be likely given the support for >= Node 18.0.0, but was something I thought of while reviewing.

Reply:
I mean, technically they could be, but since removing support for Node <18, I think we relinquish responsibility for those users, no?